### PR TITLE
FIX continue browsing button not working correctly

### LINKED ISSUE #32: Continue browsing button not working correctly.
When you update your items in your cart, the continue browsing button on top gets confused with where it is to continue to.

This was caused by issue #11  when a update cart item button was added.

### PR DESCRIPTION
**Pull request by fspringveldt**
Fixes #32.

### CODE REVIEW

File: code/controllers/DMSCartController.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Controls the `Continue browsing` link found in DMSCartNavigation.ss. Defaults all requests back to home.
+     * @return string
+     */
+    public function getContinueBrowsingLink()
+    {
+        return Director::absoluteBaseURL();
+    }
+}
```
Comment (robbieaverill) [inline code/controllers/DMSCartController.php]: what's the semantic difference between a DMSDocumentCartController and a DMSCartController? If this is intended to be a base controller for both maybe call it DMSCartAbstractController or something that indicates that?

Comment (fspringveldt) [reply code/controllers/DMSCartController.php]: Good suggestion. Yes, it is intended as a base controller. Will rename as suggested.

File: code/controllers/DMSCartAbstractController.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Controls the `Continue browsing` link found in DMSCartNavigation.ss. Defaults all requests back to home.
+     * @return string
+     */
+    public function getContinueBrowsingLink()
+    {
+        return Director::absoluteBaseURL();
+    }
+}
```
Comment (robbieaverill) [inline code/controllers/DMSCartAbstractController.php:41]: Can we have some tests for this class?

File: tests/controllers/DMSCartAbstractControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    /**
+     * Ensure the link is "friendly", not a class name
+     */
+    public function testLink()
+    {
+        $this->controller = DMSCheckoutController::create();
+        $this->assertSame('checkout', $this->controller->Link());
+        $this->assertSame('checkout/complete', $this->controller->Link('complete'));
```
Comment (robbieaverill) [inline tests/controllers/DMSCartAbstractControllerTest.php]: This behaviour belongs in the checkout controller test class - this test class should test the logic in the abstract controller

**Review by robbieaverill — CHANGES_REQUESTED** [review]

**Review by sachajudd — CHANGES_REQUESTED** [review]
When 'updating' the cart items the 'continue browsing' link should go back to the documents not back into the 'checkout' page.

File: tests/controllers/DMSDocumentCartControllerTest.php [hunk 4]
```diff
@@ -264,4 +247,13 @@ public function testView()
...
     }
+
+    /**
+     * Ensure the link is "friendly", not a class name
+     */
+    public function testLink()
+    {
+        $this->assertSame('documentcart', $this->controller->Link());
+        $this->assertSame('documentcart/view', $this->controller->Link('view'));
+    }
```
Comment (robbieaverill) [inline tests/controllers/DMSDocumentCartControllerTest.php:258]: This change seems redundant, but is just copy pasting. Please avoid in future.

**Review by robbieaverill — APPROVED** [review]
Just @sachajudd's comments. Code wise this is good.

**Review by sachajudd — APPROVED** [review]

**Pull request merged**